source/PlayerInfo.cpp

```diff
@@ -383,12 +383,24 @@ void PlayerInfo::AddChanges(list<DataNode> &changes)
 	if(changedSystems)
 	{
 		// Recalculate what systems have been seen.
-		GameData::UpdateNeighbors();
+		GameData::UpdateSystems();
 		seen.clear();
+		// All systems within the default neighbor distance of visited systems
+		// are considered seen, but if the player has an increased jump range
+		// then all systems within that jump range are considered seen. But,
+		// if a system has a static jump range less than the default, then it
+		// also has a restricted view range.
+		double jumpRange = System::DEFAULT_NEIGHBOR_DISTANCE;
+		if(flagship)
+		{
+			double flagshipRange = flagship->JumpRange();
+			if(flagshipRange > jumpRange)
+				jumpRange = flagshipRange;
+		}
 		for(const System *system : visitedSystems)
 		{
 			seen.insert(system);
-			for(const System *neighbor : system->Neighbors())
+			for(const System *neighbor : system->Neighbors(jumpRange))
 				seen.insert(neighbor);
```

Comment:
Some things that just wondered about:
- Is seen also updated/extended at the moment I install a new Jump-drive with a higher range?
- Is this updated for all previously visited systems, or only for the current one? (And what would we prefer?)

Reply:
In the case of PlayerInfo::AddChanges, this is only a one time thing after an event changes systems. It updates the seen systems using your jump range at the time of the change for all visited systems. Changing your jump range after this has occurred doesn't change which systems are seen.
PlayerInfo::Visit on the other hand updates the seen systems from your current jump range upon entering a system (not when departing a planet), buying a map, having an event set a system as seen, or loading a pilot.

This means that if you visited a system when your jump range was 100, then later gain a jump range of 200, the seen systems from the system that you only ever visited with 100 updates to show the systems within 200 whenever AddChanges detects a system change, or when the player reloads the pilot. This also means that changing your jump drive does not update the seen systems for the current system until you reenter it (although the map panel will update with the proper jump range each time you change your JD).

I believe I could easily change the seen systems to also update on planet departure so that changing your JD and then departing shows the proper seen systems (although I'd have to look somewhere else to make it so that seen systems update as soon as your jump range changes), but to only update seen systems with the jump range that the player had when they last entered the system would be more involved (and it may be the case that we don't even want to do that so that an event change can immediately reveal systems across the galaxy instead of requiring the player to travel there with an updated range).

Reply:
Isn't it lored that seen systems are observed via parallax measurements? If so, then seen should always be based solely on the default neighbor distance and hyperspace links, no matter what the jump drive range is.

Reply:
Where is that “lored”? Searching for “parallax” returns no results, and I can’t even find any references to “hyperspace” that mentions how systems become seen.

Reply:
It comes from the player's manual: https://github.com/endless-sky/endless-sky/wiki/PlayersManual#interstellar-travel

Having a larger jump range than visible range would make it possible to gate discovering a system behind having a mission active to that system (since active mission destination systems appear in the map screen even when you have no known route to it). After visiting the system, you would then be able to return to it without an active mission.

Reply:
Well using parallax to get the location of systems would imply that you need to visit two nearby systems to actually "see" a system, not just one as is the case right now, so that explanation doesn't make much sense as is. And even if we stick with the parallax explanation, what's stopping a longer range jump drive from including better measurement tools for locating systems?

Reply:
Even though we have only visited one system, we have used parallax to compute the positions of a great number of other systems. (That the game does not instantly reveal the whole map to everyone who connects their navigation computer to the galactic web, vs just those nearby systems, I don't think we will ever really explain in lore.)

I don't think we should drive jump range and system discovery from the same attribute, because it limits creative possibilities. Exactly as you suggest, what's to say some different drive doesn't have better sensors? So rather than use "jump range" for that too, we should use a different attribute, e.g. "parallax sensitivity" or "jump sensor range," whose value is used for computing seen neighbors from the current system.

Even if we don't allow modifying the discovery range, I believe everything we could do with a larger jump+discovery range can be done with just the jump range boost + clever use of missions, but not vice-versa.

Reply:
Well it's not too difficult a change, so I'm fine with leaving the view range as is.